base_tier_validation_helper/models/tier_definition_helper.py

```diff
@@ -0,0 +1,230 @@
+                    rec._prepare_tier_definition())
+                rec.tier_definition_id = tier_definition
+        self.write({'state': 'done'})
+
+    @api.multi
+    def button_delete_tier(self):
+        self.line_ids.mapped('tier_definition_id').unlink()
+        self.write({'state': 'cancel'})
+
+
+class TierDefinitionHelperLine(models.Model):
+    _name = 'tier.definition.helper.line'
+    _description = 'Tier Definition Helper Line'
+
+    parent_id = fields.Many2one(
+        comodel_name='tier.definition.helper',
+        index=True,
+        required=True,
+        ondelete='cascade'
+    )
+    approval_level = fields.Integer(
+        string='Level',
+        default=1,
+        required=True,
+    )
+    name = fields.Char(
+        string='Name',
+        default='New Tier Definition',
+    )
+    reviewer_id = fields.Many2one(
+        comodel_name="res.users",
+        string="Reviewer",
+    )
+    reviewer_group_id = fields.Many2one(
+        comodel_name="res.groups",
+        string="Reviewer group",
+    )
+    definition_domain = fields.Char(
+        string='Domain',
+        help='Expressions can be any valid python expressions.'
+             '(eg ["|", ["id", "=", 1], ["id", "=", "2"]])',
+    )
+    tier_definition_id = fields.Many2one(
+        comodel_name='tier.definition',
+        string='Tier Definition',
+        readonly=True,
+    )
+
+    def _prepare_tier_definition(self):
+        tier_definition = {
+            'model_id': self.parent_id.model_id.id,
+            'sequence': 100 - self.approval_level,
+            'review_type': self.parent_id.review_type,
+            'definition_type': self.parent_id.definition_type,
+            'notify_on_create': self.parent_id.notify_on_create,
+            'has_comment': self.parent_id.has_comment,
+            'approve_sequence': self.parent_id.approve_sequence,
+        }
+        # add name
+        if self.parent_id.auto_name:
+            tier_definition['name'] = self.parent_id.model_id.name +\
+                ' (' + str(self.approval_level) + ')'
+        else:
+            tier_definition['name'] = self.name
+        # add reviewer by type
+        if self.parent_id.review_type == 'individual':
+            tier_definition['reviewer_id'] = self.reviewer_id.id
+        elif self.parent_id.review_type == 'group':
+            tier_definition['reviewer_group_id'] = self.reviewer_group_id.id
+        # add domain
+        if self.parent_id.definition_type == 'domain':
+            tier_definition['definition_domain'] = self.definition_domain
+        return tier_definition
+
+    @api.onchange('approval_level')
+    def _onchange_approval_level(self):
+        if self.tier_definition_id:
+            tier_definition = self.env['tier.definition'].search(
+                [('id', '=', self.tier_definition_id.id)], limit=1)
+            tier_definition.write({'sequence': 100 - self.approval_level})
```

Comment:
What number 100 is from? What if approval_level is > 100?
Or can we find max(approval_level) and make the number +1, etc...

Reply:
Fixed